src/common/survival_util.cc

```diff
@@ -0,0 +1,255 @@
+#include <dmlc/registry.h>
+#include <cstdlib>
+#include <xgboost/enum_class_param.h>
+#include "survival_util.h"
+
+namespace xgboost {
+namespace common {
+
+DMLC_REGISTER_PARAMETER(AFTParam);
+
+AFTDistribution* AFTDistribution::Create(AFTDistributionType dist) {
+  switch (dist) {
+   case AFTDistributionType::kNormal:
+    return new AFTNormal;
+   case AFTDistributionType::kLogistic:
+    return new AFTLogistic;
+   case AFTDistributionType::kExtreme:
+    return new AFTExtreme;
+   default:
+    LOG(FATAL) << "Unknown distribution";
+  }
+  return nullptr;
+}
+
```

Comment:
@avinashbarnwal would be great to add a comment with reference to where these formulas come from.
e.g. survival package manual http://members.cbio.mines-paristech.fr/~thocking/survival.pdf (you might want to ask @therneau if there is a better link we can use)
and (quoted from ?survival::survreg in R)
```
All the distributions are cast into a location-scale framework,
     based on chapter 2.2 of Kalbfleisch and Prentice. 
```

Reply:
Yes. I will add relevant comments.